dubbo-config/dubbo-config-spring/src/main/java/com/alibaba/dubbo/config/spring/beans/factory/annotation/ReferenceBeanBuilder.java

```diff
@@ -80,14 +86,30 @@ private void configureConsumerConfig(Reference reference, ReferenceBean<?> refer
 
     @Override
     protected ReferenceBean doBuild() {
-        return new ReferenceBean<Object>(annotation);
+        return new ReferenceBean<Object>();
     }
 
     @Override
-    protected void preConfigureBean(Reference annotation, ReferenceBean bean) {
+    protected void preConfigureBean(Reference reference, ReferenceBean referenceBean) {
         Assert.notNull(interfaceClass, "The interface class must set first!");
+        DataBinder dataBinder = new DataBinder(referenceBean);
+        // Set ConversionService
+        dataBinder.setConversionService(getConversionService());
+        // Ignore those fields
+        String[] ignoreAttributeNames = of("application", "module", "consumer", "monitor", "registry");
+//        dataBinder.setDisallowedFields(ignoreAttributeNames);
+        // Bind annotation attributes
+        dataBinder.bind(new AnnotationPropertyValuesAdapter(reference, applicationContext.getEnvironment(), ignoreAttributeNames));
     }
 
+    private ConversionService getConversionService() {
+        DefaultConversionService conversionService = new DefaultConversionService();
+        conversionService.addConverter(new StringArrayToStringConverter());
+        conversionService.addConverter(new StringArrayToMapConverter());
```

Comment:
Is it possible to convert other scenes ?

Reply:
It's used to convert value of attributes of `@Service` and `@Reference` to required type.

Reply:
For example  , `@Service#parameters()` will be converted to `Map`